app/helpers/form_helper.rb

```diff
@@ -236,18 +236,7 @@ def byte_size_f(f, attr, options = {})
   end
 
   def counter_f(f, attr, options = {})
-    options[:class] = options[:class].to_s + ' counter_spinner' unless options[:disabled]
-    options[:help_block] ||= soft_limit_warning_block
-
-    text_f(f, attr, options)
-  end
-
-  def soft_limit_warning_block
-    content_tag(:span, :class => 'maximum-limit hidden') do
-      icon_text('warning-triangle-o',
-        content_tag(:span, ' ' + _('Specified value is higher than recommended maximum'), :class => 'error-message'),
-        :kind => 'pficon')
-    end
+    react_form_input('counter', f, attr, options)
```

Comment:
we use `counter_f` in other plugins:
https://github.com/search?q=org%3Atheforeman++counter_f&type=code
is there any regression expected from this? (I've found a regression regarding the `step` prop. as you can see down below)
would it be safer to add a deprecation warning and create a `counter_f_react` for now?

Reply:
I don't believe so, we should make sure we keep the same API, but keeping the same helper is very benefitial for the plugins - they don't need to change the usage.
Regressions are always possible, though we should try our best and if plugins find some, they should fix it in here.
If we create another helper, we would force them to switch to that helper and then switch back to the previous name.
I believe the regressions are simple to solve, thus we should be fine fixing them as they come IMHO.